My first pull request! Bot should be working without problem.

jee-gamer opened the pull request:
(is this pull request right???)
(hopefully no bugs, but well there probably will be some)
You must have a lot to say about this since I don't know if I'm doing it the right way or not.

jee-gamer committed:
Changed formal of createdAt and updated At to integer

(most basic and is easy to work with)

tried editing table in query

jee-gamer committed:
added 4 fruit vote number in database

jee-gamer committed:
trying to create Lottery

jee-gamer committed:
created lottery table from my understanding

jee-gamer committed:
Made dockerfile for database

edit query a bit to avoid errors

evd0kim committed:
relations

jee-gamer committed:
made Bet table and changes some User and Lottery table

evd0kim committed:
relations fix

evd0kim committed:
flask app added

jee-gamer committed:
Tried to create get_users_vote function
doesn't work yet

jee-gamer committed:
Made a working /users and /users_vote

fixed \n to <br> for html code to separate line

fix the error of loading twice (user_reloader)
it adds info two times making it causes error because the id isn't unique

jee-gamer committed:
added html file in templates, working POST method

you can add user from website now, you can only change the id though.

it error when refresh because it said it will use same information and go to POST method so the
same information is added again so the id is duplicate.

jee-gamer committed:
added lottery route to see all lottery

jee-gamer committed:
added message.html and adjust render_template

evd0kim committed:
api drafted

jee-gamer committed:
w

jee-gamer committed:
Merge branch 'database' of github.com:jee-gamer/Loterry

# Conflicts:
#	Database/app.py

jee-gamer committed:
added all api route

jee-gamer committed:
added list of the function the bot will need
(I have no idea how to implement it)

jee-gamer committed:
added list of the function the bot will need
(I have no idea how to implement it)

evd0kim committed:
api drafted

evd0kim committed:
query example

jee-gamer committed:
Tried to add time_left function in bot.py getting data from database

jee-gamer committed:
made working timeLeft function (not really a function now)

jee-gamer committed:
made working timeLeft function (yes)

jee-gamer committed:
made winning_fruit function to get winning_fruit from lottery
//working but incomplete, need to see how to get from specific lottery

jee-gamer committed:
Fix time_left function to work properly

jee-gamer committed:
added beta winner api route

working on getting specific lottery

evd0kim committed:
swagger draft

evd0kim committed:
refactored for connexion and openapi

evd0kim committed:
small refactoring

jee-gamer committed:
make post_user work flawlessly (I think)
added get and post user_vote

jee-gamer committed:
made get_lotery and start_lottery with unique id

jee-gamer committed:
get timeLeft + get winningFruit added [+ Testing and checks]

change lottery table to have givenTime object

jee-gamer committed:
early adaptation of bot to work with the api

jee-gamer committed:
MAJOR MAJOR Changes (damnit all I forgot to commit)

most of the bot function is working except the callback vote is still in progress.
added bot description
changing the standard lottery check sequence in majority of the bot (those like: if timeLeft < 0
or something)

make bot get id of the running lottery in every command
because get_timeLeft now requires lotteryId to function

added a lot function and path is user and swagger
including
-get_winners
-get_running_lottery
-stop_lottery

major note is that these only work if there's only one or less running lottery
(it would be very hard to make for multiple lotteries running at once)

jee-gamer committed:
finished the bot, callback_vote is working

(it's pretty slow on my end)
on your end it's pretty fast

evd0kim commented on Database.dockerfile:
```diff
@@ -0,0 +1,17 @@
+
+RUN mkdir -p "/app"
+COPY Database/database/model.py /app
+# uhh use query or model??
```
query.py was merely a test script

jee-gamer replied:
I see

evd0kim commented on Database/openapi/swagger.yaml:
```diff
@@ -0,0 +1,290 @@
+          '201':
+            description: Created
+
+    /lottery/timeLeft:
```
mixed styles for api endpoints. `timeLeft` is camel case while `users_vote` is snakecase.

evd0kim replied:
Test

evd0kim commented on DiscordBotFile/discordBot.py:
```diff
@@ -186,19 +204,24 @@ async def on_reaction_add(reaction, user):
             if sameFruitVote == 1:
-                response = f"You can vote up to {maxVote} fruit" \
-                           f"{user.name}, you already voted this fruit!"
+                response = (
```
Why are there parentheses?

jee-gamer replied:
fixed

evd0kim commented on bot.py:
```diff
@@ -31,6 +37,84 @@
+    }
+
+    async with aiohttp.ClientSession() as session:
+        async with session.post(f"{DATABASE_URL}/users_vote", json=vote) as response:
```
It makes more sense to change users_vote on something else for example usersVote or just votes

jee-gamer replied:
I made everyone endpoint snake case now so I'll keep using users_vote

evd0kim replied:
Endpoints are specific, they are not really related to codestyle of Python. These are general links. Using symbols like `_` is generally not welcomed because they are making hard reading URI. Preferred way sometimes introducing new paths like `/users/vote` assuming that there are many different endpoints under `users`.

jee-gamer replied:
I see, I'll look into that and change it

jee-gamer replied:
introduced new paths and made them camelcase

evd0kim commented on Database/api/users.py:
```diff
@@ -0,0 +1,163 @@
+    return {'message': 'Stopped a running lottery'}
+
+
+def reset_everything():  # guess this thing only reset the internet
```
Think about security. Is it really good to have such endpoint? May be it could be better to implement special requests for specific items in the tables?

jee-gamer replied:
Ok I remove that endpoint
We already have such endpoint for requesting specific item like a single user bet or a lottery so I think it's ok.

jee-gamer committed:
made a lottery_timer which tracks running lottery
it now knows if the running lottery have ended

no further function for now

jee-gamer committed:
I tried to run timer with bot (IT DOESNT WORK)

AAAAAAAAAAAAAAAAAAAAAAAAAA
it always say

RuntimeError: This event loop is already running
sys:1: RuntimeWarning: coroutine 'Executor._startup_polling' was never awaited
sys:1: RuntimeWarning: coroutine 'Executor._shutdown_polling' was never awaited

jee-gamer committed:
Made working timer.
runs every time a lottery is created

fix some bugs in bot replies.

evd0kim commented:
Test comment

evd0kim commented on Database/api/users.py:
```diff
@@ -0,0 +1,163 @@
+    return jsonify(lotteryId)
+
+
+def stop_lottery():
```
Think about security. Is it really good to have such endpoint? May be it could be better to implement special requests for specific items in the tables?

jee-gamer replied:
really..?
but this one is stop_lottery()
and we also have the start_lottery() one that can start the lottery.

if you really mean security in stop_lottery() then I dont know what I can do

evd0kim replied:
After your commits strings moved a bit so instead of endpoint for resetting whole database you see this.

evd0kim committed:
notification worker

jee-gamer committed:
added lottery.py

jee-gamer committed:
added lottery.py

jee-gamer committed:
start_lottery() now returns lottery info if it did start one

so now you can take the givenTime from there, like what you wanted

jee-gamer committed:
bruh timer is running forever

jee-gamer committed:
Tried making backendClient

jee-gamer committed:
Made backendClient and make it work with bot

I dont know if this is correct

jee-gamer commented:
I was testing

jee-gamer committed:
Made working timer that run every 5 seconds
to check if there is a running lottery and if the time is up
when the time is up, stop the lottery and send messages to subscriber.

then keep running the check.

still doesn't work simultaneously with bot T-T

jee-gamer committed:
Tidy up the backendClient

evd0kim committed:
run_coroutine_threadsafe for notify

evd0kim commented on bot.py:
```diff
@@ -212,4 +240,11 @@ async def message_not_modified_handler(update, error):
 if __name__ == "__main__":
+    print("Launching Timeout Worker")
+    loop = asyncio.get_event_loop()
+    asyncio.run_coroutine_threadsafe(timer.notify(), loop)
```
Here is how it should be running co-routines together. 

I ran the lottery but the bot didn't respond with anything when it finished while I see that it should send either winning message or not-winning anyone message.

jee-gamer replied:
It works! Thank you!

"I ran the lottery but the bot didn't respond with anything when it finished while I see that it should send either winning message or not-winning anyone message."

It works for me when I tested it right now

evd0kim commented on old_lottery_class.py:
This needs to be removed and lottery_timer needs to be refactored. It was created just for fast demo. Now it is time to get code into some shape. There is long trail of old_lottery_class names. It is not good.

jee-gamer replied:
removed.

jee-gamer replied:
I think the lottery_timer is in a pretty minimal state now, and the 2 new functions are made specifically for its purpose. so I don't know how to refactor it.
can you give some insight about it? I can probably make it even more minimal but it'll be more complex and harder to understand and fix.

evd0kim replied:
No, you made a good explanation. I get your point. I need to look at timer again to continue.

evd0kim commented on Database/backendClient.py:
Database Client should be placed in separate directory since Database is actually a web server with embedded SQLite database.

jee-gamer replied:
moved to a new BackendClient directory under Loterry, is that ok?

evd0kim replied:
Yes, It is fine.

jee-gamer committed:
fixed the mixed style problem in end point, got rid of the old lottery class

jee-gamer committed:
minimal changes (I dont know what I did)

jee-gamer committed:
moved backendClient from Database to new directory under Loterry

jee-gamer committed:
introduced new paths and made them camelcase

jee-gamer committed:
Added endpoint to get all bet
Now the notify subscriber is applied to the person that votes on this lottery only

jee-gamer committed:
made bitcoin request app
